File: API/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
        public long Price { get; set; }
        public int QuantityInStock { get; set; }
        public string Genre { get; set; }
        public string Theme { get; set; }
        public string Developer { get; set; }
        public DateOnly ReleaseDate { get; set; }
    }
}

File: client/src/app/models/product.ts
export interface Product {
    id: number;
    name: string;
    description: string;
    image: string;
    price: number;
    quantityInStock: number;
    genre: string;
    theme: string;
    developer: string;
    releaseDate: Date;
}

export interface ProductParams {
    orderBy: string;
    searchTerm?: string;
    genres: string[];
    themes: string[];
    pageSize: number;
    pageNumber: number;
}